shishito/library/modules/runtime/environment/shishito_environment.py

```diff
@@ -1,28 +1,114 @@
-from shishito.library.modules.runtime.shishito_support import ShishitoSupport
+import os
+from selenium import webdriver
+import sys
 
 
 class ShishitoEnvironment(object):
-    def __init__(self):
-        self.shishito_support = ShishitoSupport()
+
+    def __init__(self, shishito_support):
+        self.shishito_support = shishito_support
         self.capabilities = None
 
+        self.platform_name = self.shishito_support.gid('test_platform')
+        self.environment_name = self.shishito_support.gid('test_environment')
+
+        self.config = self.shishito_support.get_environment_config(
+            platform_name=self.platform_name,
+            environment_name=self.environment_name
+        )
+
+        self.project_root = os.getcwd()
+
+    def add_extension_to_browser(self, browser_type, browser_profile):
+        """ returns browser profile updated with one or more extensions """
+
+        if browser_type == 'chrome':
+            all_extensions = self.get_extension_file_names('crx')
+            for chr_extension in all_extensions:
+                browser_profile.add_extension(os.path.join(self.project_root, 'extension', chr_extension + '.crx'))
+        elif browser_type == 'firefox':
+            all_extensions = self.get_extension_file_names('xpi')
+            for ff_extension in all_extensions:
+                browser_profile.add_extension(os.path.join(self.project_root, 'extension', ff_extension + '.xpi'))
+
+        return browser_profile
+
     def call_browser(self, combination, capabilities=None):
         """ Starts browser """
+
         # get browser capabilities and profile
         if capabilities:
             capabilities.update(self.get_capabilities(combination))
         else:
             capabilities = self.get_capabilities(combination)
-        return self.start_driver(combination, capabilities)
+
+        # get browser type
+        browser_type = self.config.get(combination, 'browser')
+        browser_type = browser_type.lower()
+
+        # get driver
+        driver = self.start_driver(browser_type, capabilities)
+
+        # set browser size is defined
+        browser_size = self.config.get(combination, 'resolution')
+        if browser_size:
+            # default size --> leave it on webdriver
+            width, height = browser_size.split('x')
+            driver.set_window_size(width, height)
+
+        return driver
 
     def get_capabilities(self, combination):
         """ Returns dictionary of browser capabilities """
         pass
 
-    def start_driver(self, combination, capabilities):
+    def start_driver(self, browser_type, capabilities, remote_driver_url=None):
         """ Starts driver """
-        pass
+
+        browser_profile = self.get_browser_profile(browser_type, capabilities)
+
+        if remote_driver_url is None:
```

Comment:
default value is None and raises error? then remove default value from method signature